commands/build.go

```diff
@@ -643,6 +652,101 @@ func parsePrintFunc(str string) (*build.PrintFunc, error) {
 	return f, nil
 }
 
+func addGitProvenance(ctx context.Context, labels []string, contextPath string, dockerfilePath string) ([]string, error) {
+	v, ok := os.LookupEnv("BUILDX_GIT_LABELS")
+	if !ok || contextPath == "" {
+		return labels, nil
+	}
+
+	// figure out in which directory the git command needs to run in
+	var wd string
+	if filepath.IsAbs(contextPath) {
+		wd = contextPath
+	} else {
+		cwd, _ := os.Getwd()
+		wd, _ = filepath.Abs(filepath.Join(cwd, contextPath))
+	}
+
+	// obtain Git sha of current HEAD
+	cmd := exec.CommandContext(ctx, "git", "rev-parse", "HEAD")
+	cmd.Dir = wd
+	out, err := cmd.Output()
+	if err != nil {
+		return nil, errors.Errorf("error obtaining git head: %v", err)
+	}
+	sha := strings.TrimSpace(string(out))
+
+	// check if the current HEAD is clean
+	cmd = exec.CommandContext(ctx, "git", "status", "--porcelain", "--ignored")
+	cmd.Dir = wd
+	out, err = cmd.Output()
+	if err != nil {
+		return nil, errors.Errorf("error obtaining git status: %v", err)
+	}
+	if len(strings.TrimSpace(string(out))) == 0 {
+		labels = append(labels, fmt.Sprintf("%s=%s", annotations.AnnotationRevision, sha))
+	} else {
+		labels = append(labels, fmt.Sprintf("%s=%s-dirty", annotations.AnnotationRevision, sha))
+	}
+
+	// add a remote url if full Git details are requested; if there aren't any remotes don't fail
+	if v == "full" {
+		cmd = exec.CommandContext(ctx, "git", "remote")
```

Comment:
I *think* we can use `git ls-remote --get-url` to get the URL of the current remote: https://git-scm.com/docs/git-ls-remote#Documentation/git-ls-remote.txt---get-url, so we shouldn't need to interact with `git config` manually.

Reply:
Done